### PR TITLE
Promote KEP-2681 to GA in 1.30

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Promoted the `status.hostIPs` field for Pods to general availability.
The `PodHostIPs` feature gate no longer has any effect, and the
`status.hostIPs` field is always available within the Pod API.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/2681
- [Usage]: <link>
- [Other doc]:
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1071,7 +1072,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	PodReadyToStartContainersCondition: {Default: true, PreRelease: featuregate.Beta},
 
-	PodHostIPs: {Default: true, PreRelease: featuregate.Beta},
+	PodHostIPs: {Default: true, PreRelease: featuregate.GA}, // GA in 1.30; Lock to default in 1.31; remove in 1.32
```
Comment (danwinship) [inline pkg/features/kube_features.go:1075]: we normally lock to default when it goes to GA... is there any reason to not do that here?

(and you can't remove the tag from the e2e test if it's still possible to disable the feature)

Comment (wzshiming) [reply pkg/features/kube_features.go]: Done

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -1087,7 +1088,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	PodReadyToStartContainersCondition: {Default: true, PreRelease: featuregate.Beta},
 
-	PodHostIPs: {Default: true, PreRelease: featuregate.Beta},
+	PodHostIPs: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.31
```
Comment (danwinship) [inline pkg/features/kube_features.go:1091]: sorry, the "remove" should still be 1.32; we leave them around for 2 releases after GA

File: pkg/api/pod/util_test.go [hunk 3]
```diff
@@ -1179,81 +1179,42 @@ func TestDropDisabledPodStatusFields(t *testing.T) {
...
+		podStatus     *api.PodStatus
+		oldPodStatus  *api.PodStatus
+		wantPodStatus *api.PodStatus
 	}{
 		{
-			name:           "gate off, old=without, new=without",
-			oldPodStatus:   podWithoutHostIPs(),
-			podStatus:      podWithoutHostIPs(),
-			featureEnabled: false,
+			name:         "gate on, old=without, new=without",
```
Comment (danwinship) [inline pkg/api/pod/util_test.go:1188]: remove the references to the feature gate in the test case names

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by sftim** [comment]
Changelog suggestion
```diff
-Promote PodHostIPs condition to GA and lock to default.
+Promoted the `status.hostIPs` field for Pods to general availability.
+The `PodHostIPs` feature gate is now enabled and locked to active.
```

**Comment by wzshiming** [comment]
Done, thanks.

**Pull request merged**